lib/CFRPageActions.jsm

```diff
@@ -275,6 +276,52 @@ class PageAction {
     }
   }
 
+  async _renderPinTabAnimation() {
+    const footer = this.window.document.getElementById("cfr-notification-footer");
+    const ANIMATION_CONTAINER_ID = "cfr-notification-footer-pintab-animation-container";
+    let animationsEnabled = Services.prefs.getBoolPref(ANIMATIONS_ENABLED_PREF, true);
+    let animationContainer = this.window.document.getElementById(ANIMATION_CONTAINER_ID);
+    if (animationContainer) {
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+      animationContainer.removeAttribute("paused");
+    } else {
+      animationContainer = this.window.document.createElement("vbox");
+      animationContainer.setAttribute("id", ANIMATION_CONTAINER_ID);
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+
+      let spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
+      animationContainer.appendChild(spacer);
+
+      let controlsContainer = this.window.document.createElement("hbox");
+      controlsContainer.setAttribute("id", "cfr-notification-footer-animation-controls");
+
+      spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
+      controlsContainer.appendChild(spacer);
+
+      let pauseButton = this.window.document.createElement("hbox");
+      pauseButton.setAttribute("id", "cfr-notification-footer-pause-button");
+      pauseButton.addEventListener("click", () => {
```

Comment:
We need to remove this event listener.

Reply:
Probably in `_popupStateChange`

Reply:
I just added it on `window.unload`.

Reply:
I don't think this is strictly necessary as-is considering the event listener gets added only on element creation and presumably the element gets destroyed on `window.unload` anyway, although I wonder if it would be more appropriate to attach/remove the event listener on `_expand`/`_collapse` so that it's not long-lived throughout the whole life of the window. What do you think?